test/Dialect/FIRRTL/remove-reset.mlir

```diff
@@ -0,0 +1,126 @@
+// RUN: circt-opt --pass-pipeline='firrtl.circuit(firrtl-lower-types,firrtl-imconstprop),canonicalize' %s | FileCheck %s
```

Comment:
It's great we're adding these tests from the SFC testsuite, but it is a bit unconventional that we're running several tightly coupled passes in a fixed ordering as part of the test.  This isn't ideal as an integration test, because the fixed list will drift from firtool, and isn't ideal as a unit test because it isn't testing individual passes.

If these want to be unit tests, I'd recommend spliting them up and putting the pieces in the existing .mlir files for the various passes (you can attribute where each circuit came from on a case by case basis).  If you'd like to do integration tests for these, then I'd recommend we add something like a `test/Dialect/FIRRTL/SFCTests/` directory and check these in as .fir files.

WDYT?

-Chris

I think these want to be integration tests

Reply:
> I think these want to be integration tests

Yeah this sounds like a great idea. Let me move these over into a separate directory in a follow-up PR!